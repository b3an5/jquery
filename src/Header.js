import React from 'react'

export default function Header(props) {
  return(
    <div className='header'>
      <h1 className='main-title header-content'>Jquery Quiz</h1>
      <h2 className='score header-content'>Score:<span>{props.score}</span></h2>
    </div>
  )
}