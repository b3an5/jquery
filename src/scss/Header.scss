.header {
  position: fixed;
  padding-bottom: 5px;
  z-index: 2;
  width: 100vw;
  background-color: aqua 
}

.score {
  position: absolute;
  top: 27px;
  right: 5px;
}