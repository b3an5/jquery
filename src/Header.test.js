import React from 'react';
import Header from './Header.js';
import { shallow } from 'enzyme';

describe('Header', () => {

  let wrapper;
  const mockResetGame = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <Header 
      resetGame={mockResetGame} />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should run showMainPage when reset game is pressed', () => {
    wrapper.find('.reset-button').simulate('click')
    expect(mockResetGame).toBeCalled();
  })

})