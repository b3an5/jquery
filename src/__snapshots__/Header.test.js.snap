// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    resetGame={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="main-title header-content"
        >
          Jquery Quiz
        </h1>,
        <button
          className="question-button reset-button"
          onClick={[MockFunction]}
        >
          Reset Game
        </button>,
        <h2
          className="score header-content"
        >
          Score:
          <span />
        </h2>,
        <h2
          className="high-score header-content"
        >
          High Score:
          <span />
        </h2>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Jquery Quiz",
          "className": "main-title header-content",
        },
        "ref": null,
        "rendered": "Jquery Quiz",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Reset Game",
          "className": "question-button reset-button",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "Reset Game",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Score:",
            <span />,
          ],
          "className": "score header-content",
        },
        "ref": null,
        "rendered": Array [
          "Score:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "High Score:",
            <span />,
          ],
          "className": "high-score header-content",
        },
        "ref": null,
        "rendered": Array [
          "High Score:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": null,
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
        ],
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="main-title header-content"
          >
            Jquery Quiz
          </h1>,
          <button
            className="question-button reset-button"
            onClick={[MockFunction]}
          >
            Reset Game
          </button>,
          <h2
            className="score header-content"
          >
            Score:
            <span />
          </h2>,
          <h2
            className="high-score header-content"
          >
            High Score:
            <span />
          </h2>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Jquery Quiz",
            "className": "main-title header-content",
          },
          "ref": null,
          "rendered": "Jquery Quiz",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reset Game",
            "className": "question-button reset-button",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Reset Game",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Score:",
              <span />,
            ],
            "className": "score header-content",
          },
          "ref": null,
          "rendered": Array [
            "Score:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "High Score:",
              <span />,
            ],
            "className": "high-score header-content",
          },
          "ref": null,
          "rendered": Array [
            "High Score:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": null,
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
